### PR TITLE
Replace `useComposeRefs` with `useCombinedRefs`

### LINKED ISSUE #10093: Code Quality: Consolidate useCombinedRefs & useComposeRefs usage
These two hooks seem to duplicate the same functionality. This ticket is to just use one of them and replace the other everywhere else.

https://github.com/google/web-stories-wp/blob/main/packages/react/src/useCombinedRefs.js

https://github.com/google/web-stories-wp/blob/main/packages/design-system/src/utils/useComposeRefs.js

Probably best to keep whichever hook we consolidate too inside the `/react` package because of the current dependency setup.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

These two hooks seem to duplicate the same functionality. Remove useComposeRefs

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10093

### CODE REVIEW

File: packages/react/src/test/useComposeRefs.js [hunk 1]
```diff
@@ -16,14 +16,14 @@
...
 import { renderHook } from '@testing-library/react-hooks';
 /**
  * Internal dependencies
  */
-import useComposeRefs from '../useComposeRefs';
+import { useRef } from '..';
+import useCombinedRefs from '../useCombinedRefs';
 
-describe('useComposeRefs', () => {
+describe('useCombinedRefs', () => {
```
Comment (swissspidy) [inline packages/react/src/test/useComposeRefs.js]: Please also rename the file to `useCombinedRefs.js`

Comment (spacedmonkey) [reply packages/react/src/test/useComposeRefs.js]: I thought I did.

**Review by swissspidy — APPROVED** [review]

**Comment by maxyinger** [comment]
@spacedmonkey just noticed this with `useCombineRefs` implementation, but it will create a callback ref on every render because a rest parameter returns a new array instance every time it is called. Made a little codesandbox to visualize:
https://codesandbox.io/s/patient-grass-b66b8?file=/src/index.js

I believe the `useComposeRefs` implementation avoided this while maintaining the same functionality. Sorry for not catching this earlier.

Probably doesn't have too many implications, but probably good just for code quality sake

**Comment by swissspidy** [comment]
@littlemilkstudio could you perhaps create quick follow-up PR?

**Comment by maxyinger** [comment]
follow-up here:
https://github.com/google/web-stories-wp/pull/10183

**Pull request merged**